actions/storage.js

```diff
@@ -84,22 +84,31 @@ export function actionOnItemsWithPrefix(prefix, action) {
 
 // Temporary action to manually rehydrate drafts from localStorage.
 function rehydrateDrafts() {
-    return (dispatch) => {
+    return (dispatch, getState) => {
         const actions = [];
 
-        Object.entries(localStorage).forEach((entry) => {
-            const key = entry[0];
-            const value = entry[1];
-            if (key.indexOf(StoragePrefixes.DRAFT) === 0 || key.indexOf(StoragePrefixes.COMMENT_DRAFT) === 0) {
-                actions.push({
-                    type: StorageTypes.SET_GLOBAL_ITEM,
-                    data: {name: key, value: JSON.parse(value), timestamp: new Date()},
-                });
+        const state = getState();
+
+        for (const [key, value] of Object.entries(localStorage)) {
+            if (!key.startsWith(StoragePrefixes.DRAFT) && !key.startsWith(StoragePrefixes.COMMENT_DRAFT)) {
+                continue;
             }
-        });
 
-        dispatch(batchActions(actions));
-        return {data: true};
+            const parsed = JSON.parse(value);
+
+            const existing = getGlobalItem(state, key);
+            if (existing && areObjectsEqual(existing, parsed)) {
+                continue;
+            }
+
+            actions.push(setGlobalItem(key, parsed));
+        }
+
+        if (actions.length === 0) {
+            return {data: false};
```

Comment:
nit: We never used to do this, even if there were no entries. Are there any potentially negative side effects?

Reply:
No, there's no changes from this since we never await on this dispatch. Having to return `{data: 'anything'}` is an artifact of our messed up typing that we really need to fix at some point.